### PR TITLE
CMake: Bugfix: Only set test dependencies if tests are actually defined...

### PR DESCRIPTION
**Pull request by tamiko**
Otherwise "big bada boom".

### CODE REVIEW

File: tests/a-framework/CMakeLists.txt [hunk 1]
```diff
@@ -10,14 +10,23 @@ SET(TEST_TARGET_DEBUG dummy.debug)
...
-    DEPENDS a-framework/parameter_file_1.${_build}
-    )
-ENDFOREACH()
+# We only do this for CMake-3.4 or newer that has policy CMP0064 and
+# supports the IF(TEST ...) operator.
+#
+IF(POLICY CMP0064)
+  CMAKE_POLICY(SET CMP0064 NEW)
+  FOREACH(_build ${DEAL_II_BUILD_TYPES})
+    IF( TEST a-framework/parameter_file_2.${_build} AND
```
Comment (tjhei) [inline tests/a-framework/CMakeLists.txt]: this does not need to be tolower?

File: tests/a-framework/CMakeLists.txt [hunk 2]
```diff
@@ -10,14 +10,23 @@ SET(TEST_TARGET_DEBUG dummy.debug)
...
+# Limit concurrency between the two parameter file tests.
 #
-FOREACH(_build ${DEAL_II_BUILD_TYPES})
-  STRING(TOLOWER ${_build} _build)
-  SET_TESTS_PROPERTIES(a-framework/parameter_file_2.${_build} PROPERTIES
-    DEPENDS a-framework/parameter_file_1.${_build}
-    )
-ENDFOREACH()
+# We only do this for CMake-3.4 or newer that has policy CMP0064 and
+# supports the IF(TEST ...) operator.
```
Comment (tjhei) [inline tests/a-framework/CMakeLists.txt]: why do you need this only for 3.4+ ? Can you leave a comment?

File: tests/a-framework/CMakeLists.txt (hunk 1 above)
Comment (tjhei) [reply tests/a-framework/CMakeLists.txt]: (you fixed this when I wrote the comment)

Comment (tamiko) [reply tests/a-framework/CMakeLists.txt]: Jup. Still verifying that this does the right thing in all cases.

Comment (tamiko) [reply tests/a-framework/CMakeLists.txt]: Done. Looks good on my machine now.

File: tests/a-framework/CMakeLists.txt (hunk 2 above)
Comment (tamiko) [reply tests/a-framework/CMakeLists.txt]: The problem is that a CMake before version 3.4 doesn't have the functionality in question.

**Comment by tjhei** [comment]
Much better, thanks!

File: tests/a-framework/CMakeLists.txt [hunk 3]
```diff
@@ -10,14 +10,25 @@ SET(TEST_TARGET_DEBUG dummy.debug)
...
-FOREACH(_build ${DEAL_II_BUILD_TYPES})
-  STRING(TOLOWER ${_build} _build)
-  SET_TESTS_PROPERTIES(a-framework/parameter_file_2.${_build} PROPERTIES
-    DEPENDS a-framework/parameter_file_1.${_build}
-    )
-ENDFOREACH()
+# We only do this for CMake-3.4 or newer that has policy CMP0064 and
+# supports the IF(TEST ...) operator. For older CMake versions that do not
+# support the IF(TEST ...) operator we cannot much do (and people have to
+# live with spurious test failures.)
```
[the remark this answers is not in the document]
Comment (tamiko) [reply tests/a-framework/CMakeLists.txt]: @tjhei Updated to use an internal TARGET ^^

**Pull request merged**